.github/workflows/build_test.yml

```diff
@@ -61,14 +65,38 @@ jobs:
       uses: actions/checkout@v2
       with:
         submodules: true
+        fetch-depth: 2
+    - name: Git environment
+      id: git-env
+      run: |
+        echo "::set-output name=parent::$(git rev-parse ${{ github.sha }}^)"
+      shell: bash
+    - name: ccache
+      uses: actions/cache@v2
```

Comment:
I assume that's "cache of previous commit"?

Reply:
actions/cache@v2 is the GitHub action that lets you retrieve and store "a cache", up to 5GB per repo. A "cache" has a key and it is created only once (until it is garbage collected when you run out of space). So this get the cache of key "$os_name-$my_commit-sha-$build_name" and uses it. If it doesn't exist (if the key: doesn't exist) it then tries to retrieve the one from restore-keys (you can pass multiple). If there was no cache for key: it will create one at the end of the pipeline.

Reply:
@v2 is just the release v2 of the actions/cache repo https://github.com/actions/cache which is github's own actions org.